projects/documentation/content/migrations/2023-6-11.md

```diff
@@ -0,0 +1,97 @@
+
+### Remove the `open` attribute from content that is meant to be overlaid.
+
+The `open` attribute will be addressed directly by the Overlay API itself to manage the visual transition from the closed to open state of the overlaid content, and management of this attribute by the parent application could prevent those transitions from occurring as expected.
+
+✅ _DO_ exclude `open` attributes from `slot="*-content"` children of `<overlay-trigger>` elements
+
+```html
+<overlay-trigger>
+    <sp-button slot="trigger">Trigger</sp-button>
+    <sp-popover slot="click-content" placement="bottom">
+        <sp-dialog no-divider>Popover for the trigger</sp-dialog>
+    </sp-popover>
+    <sp-tooltip slot="hover-content" placement="right">
+        Tooltip for the trigger
+    </sp-tooltip>
+    <!-- etc. -->
+</overlay-trigger>
+```
+
+### Remove usage of `placement="none"` in both declarative and imperative API usages
+
+`placement="none"` or `placement: 'none'` was previously leveraged to outline an overlay that would take the full size of the viewport, this responsibility is now fulfilled by assigning this value as `undefined`, or, better yet, not including it at all.
+
+✅ _DO_ omit `placement` when not specifically relating the overlaid content to its target with the imperative API
+
+```ts
+this.closeHoverOverlay = Overlay.open(
+    triggerReference,
+    'modal',
+    contentReference,
+    {
+        delayed: false,
+        offset: 0,
+        receivesFocus: 'auto',
+    }
+);
+```
+
+### Ensure that your consumption of `sp-opened` and `sp-closed` events are typed
+
+Some of the internal properties of these events are changing, see `interaction: 'auto' | 'hint' | 'manual' | 'modal' | 'page'`. Ensuring you are consuming these types will allow Typescript to support your upgrade from one version of the API to the next.
+
+✅ _DO_ use the `OverlayOpenCloseDetail` type when listening for these Custom Events
+
+```ts
+html`
+    <element-containing-an-overlay
+        @sp-closed=${(event: OverlayOpenCloseDetail) => {
+            if (event.detail.interaction === 'auto') {
+                // Do something when the event was dispatched for an overlay with type "auto".
+            }
+        }}
+        @sp-opened=${(event: OverlayOpenCloseDetail) => {
+            if (event.detail.interaction === 'modal') {
+                // Do something when the event was dispatched for an overlay with type "modal".
+            }
+        }}
+    ></element-containing-an-overlay>
+`;
+```
+
+### Prepare for descendant overlays to exist in the same DOM tree
+
+Previously, the reparenting of overlay content prevented `sp-opened` and `sp-closed` events from bubbling as would normally be the default. This means you will have the opportunity to hear and respond to these events ALL in a centralized location, but your listeners may hear more of these events than you expect and/or had previously.
```

Comment:
> Previously, the reparenting of overlay content prevented `sp-opened` and `sp-closed` events from bubbling, which would normally be the default. This means you will have the opportunity to hear and respond to ALL these events in a singular location, but your listeners may hear more of these events than you expect and/or had previously.

I don't know if "ALL these events" is accurate as opposed to they ALL are responded to in a single location, but I think saying singular rather than centralised makes more sense.

Reply:
Edited again to try to get your suggestions and the meaning here:
> Previously, the reparenting of overlay content prevented `sp-opened` and `sp-closed` events from propagating through the overlay's ancestor DOM tree. Going forward these events will propagate in a more native manner meaning that an ancestor will have the opportunity to hear and respond to ALL `sp-opened` and `sp-closed` events for all of its descendant overlays.